lib/Transforms/Scalar/PromoteVerifierCalls.cc

```diff
@@ -32,6 +34,41 @@ Value *coerceToBool(Value *arg, IRBuilder<> &builder, Instruction &I) {
     arg = builder.CreateICmpNE(arg, ConstantInt::get(arg->getType(), 0));
   return arg;
 }
+
+/// Determines if \p v represents a partial function call. If so, then the
+/// function call is returned.
+Value *extractPartialFnCall(Value *v) {
+  // The partial call may be cast to bool by comparison to 0.
+  if (auto cmp = dyn_cast<ICmpInst>(v)) {
+    auto *lhs = cmp->getOperand(0);
+    auto *rhs = cmp->getOperand(1);
+    v = isa<CallInst>(lhs) ? lhs : rhs;
+  }
+
+  // The value must now be a call instruction.
+  if (!isa<CallInst>(v))
+    return nullptr;
+
+  // Annotation2Metadata marks all instructions of the partial function with the
```

Comment:
why not mark function declaration only?

Reply:
This is how `Annotation2Metadata` is implemented. It reads the from `llvm.global.annotations`, extracts the function annotations, and then annotates each instruction within the function.

Reply:
you are right. I did not see this at first. I was expecting the annotation to be attached to the function only. Oh well